### PR TITLE
add debugger poc

### PR DESCRIPTION
**Pull request by mjudeikis**
Debugger script for gathering logs

cc: @openshift/sig-azure

### CODE REVIEW

**Comment by mjudeikis** [comment]
@jim-minter if you happy with content and name (`osa-debugger`) will move to `openshift-ansible`

File: helpers/debugger/osa-debugger [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+set -x
+
+trap rm -rf $LOG_DIR
```
Comment (jim-minter) [inline helpers/debugger/osa-debugger]: move after LOG_DIR is set
did you mean `trap 'rm -rf $LOG_DIR' EXIT` ?

File: helpers/debugger/osa-debugger [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+# Dump systemd services logs
+for name in ${SYSTEMD_SERVICES} ; do
+  dump_file_path=${LOG_DIR}/${name}.log
+  journalctl -u ${name}.service --since "${since_systemd}" > $dump_file_path
+done
+
+# Dump command output
+ip a > ${LOG_DIR}/ip_a.log
+netstat > ${LOG_DIR}/netstat.log
```
Comment (jim-minter) [inline helpers/debugger/osa-debugger]: `netstat -antu` ?

File: helpers/debugger/osa-debugger [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+mount > ${LOG_DIR}/mount.log
+
+for table in filter nat; do 
+    iptables -t $table -nvL > ${LOG_DIR}/iptables_$table.log
+done
+
+# Dump system journal
+journalctl --since "${since_systemd}" > ${LOG_DIR}/journalctl.log
+
+# Dump snd container logs
```
Comment (jim-minter) [inline helpers/debugger/osa-debugger]: s/snd/sdn/

File: helpers/debugger/osa-debugger [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+uid=$(docker ps -l -a --filter "label=io.kubernetes.container.name=sdn" --format '{{ .Label "io.kubernetes.pod.uid" }}')                                                                                                                    
+if [[ ! -z "${uid}" ]]; then
+  container=$(docker ps -l -a -q --filter "label=io.kubernetes.pod.uid=${uid}" --filter "label=io.kubernetes.container.name=sdn")
+  docker logs --since ${since_docker} "${container}" >& ${LOG_DIR}/sdn.log
+fi
+
+prefix=osa_node_$(hostname)_$(date +%Y%m%d%H%M%S)
+tar -czPf $prefix.tar.gz  --xform="s|^$LOG_DIR|$prefix|" $LOG_DIR
+
+trap "{ rm -rf $LOG_DIR ; exit 255; }" EXIT
```
[the remark this answers is not in the document]
Comment (mjudeikis) [reply helpers/debugger/osa-debugger]: I know there was a reason I split them, some edge case... But I cant remember :/ . Will try to dig it out again in my brains...

**Comment by mjudeikis** [comment]
close via https://github.com/openshift/openshift-ansible/pull/9057

**Pull request closed without merging**